blockonomics-woocommerce.php

```diff
@@ -423,33 +423,59 @@ function show_basic() {
         </div>
     <?php
     }
-    function bnomics_display_tx_info($order)
+    function bnomics_display_payment_details($order, $transactions, $email=false)
     {
-        global $wpdb;
-        $order_id = $order->get_id();
-        $table_name = $wpdb->prefix .'blockonomics_payments';
-        $query = $wpdb->prepare("SELECT * FROM ". $table_name." WHERE order_id = " . $order_id . " AND txid != ''");
-        $transactions = $wpdb->get_results($query,ARRAY_A);
-
-        if (empty($transactions)) {
-            return;
-        }
-
-        echo '<b>'.__('Payment Details', 'blockonomics-bitcoin-payments').'</b><br />';
-                    
+        $total_paid_fiat = 0;
+        
+        echo '<h2 class="woocommerce-column__title">Payment details</h2>';
+        echo '<table class="woocommerce-table woocommerce-table--order-details shop_table order_details"><tfoot>'; 
+        
         foreach ($transactions as $transaction) {
+            $total_paid_fiat = $total_paid_fiat + (float)$transaction['paid_fiat'];
+
             if($transaction['crypto'] == 'btc') {
                 $base_url = Blockonomics::BASE_URL;
             } else {
                 $base_url = Blockonomics::BCH_BASE_URL;
             }
-            echo '<a style="word-wrap: break-word;" href="' . $base_url . '/api/tx?txid=' . $transaction['txid'] . '&addr=' . $transaction['address'] . '">' . $transaction['txid'] . '</a>';
-            echo '<br />';
+            
+            echo '<tr><th scope="row"><b>';
+            echo '<a style="word-wrap: break-word;" href="' . $base_url . '/api/tx?txid=' . $transaction['txid'] . '&addr=' . $transaction['address'] . '">' . $transaction['txid'] . '</th></a>';
+            echo '<td>' . wc_price($transaction['paid_fiat']) . '<br /></td></tr>';
         }
 
-        echo '</p>';
-            
-         
+        $expected_fiat = (float)$order->get_total();
+
+        if ($total_paid_fiat >= $expected_fiat) {
```

Comment:
If $total_paid_fiat >= $expected_fiat, the table structure is not completed and displays incorrectly